File: apps/api/app/schemas/ai_config.py
"""AI 提供商配置 schema。"""

from pydantic import BaseModel, Field


class OpenAiConfig(BaseModel):
    """OpenAI 兼容 API 配置。"""

    api_key: str = Field(default="", description="OpenAI 或兼容服务的 API Key")
    base_url: str = Field(default="", description="可选自定义 Base URL")
    model: str = Field(default="gpt-4o-mini", description="默认模型名称")


class AiConfig(BaseModel):
    """OpenAI 兼容 API 配置。"""

    openai: OpenAiConfig = Field(default_factory=OpenAiConfig)
    # token 总量额度上限，用于「token 用量」页面计算消耗占比
    token_quota: int = Field(default=1_000_000, ge=1, description="token 总量额度")

File: apps/admin/src/types/ai-config.ts
/** OpenAI 兼容 API 配置 */
export interface OpenAiConfig {
  api_key: string;
  base_url: string;
  model: string;
}

/** 完整 AI 配置 */
export interface AiConfig {
  openai: OpenAiConfig;
  /** token 总量额度，用于「token 用量」页面计算消耗占比 */
  token_quota: number;
}

/** 新建空配置时的默认值 */
export const EMPTY_OPENAI_CONFIG: OpenAiConfig = {
  api_key: "",
  base_url: "",
  model: "gpt-4o-mini",
};

/** 完整 AI 配置默认值 */
export const EMPTY_AI_CONFIG: AiConfig = {
  openai: EMPTY_OPENAI_CONFIG,
  token_quota: 1_000_000,
};
